Scripts/Level_Components/IDE.gd

```diff
@@ -36,14 +40,39 @@ func _on_Button_pressed():
 
 
 func enter_scope(node):
+	#If current IDE block node is not a Loop
+	if !is_loop(node.name): 
```

Comment:
If (looping): //Just finished a loop, only called if condition already passed
   code = loopNode.get_code() //Don't need to actually go into a new scope, just need to get code again
ElIf (current scope is loop): //In a loop rn, but changing scope to something else
--- code.push_back(loopNode) //Save the fact that we're in a loop

//Continue OG execution